### PR TITLE
Create a kubevela demo using GoogleCloudPlatform/microservices-demo

### PR DESCRIPTION
**Pull request by yangsoon**
https://github.com/oam-dev/kubevela/issues/957

- [x]  Create WorkloadDefinition and TraitDefinition
- [x]  Improve documentation

### CODE REVIEW

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
+# GoogleCloudPlatform-MicroServices-Demo with Kubevela
+
+[MicroServices-Demo](https://github.com/GoogleCloudPlatform/microservices-demo) is a cloud-native microservices demo application.
+This Demo create a **Online Boutique** application. **Online Boutique** consists of a 10-tier microservices application. 
+The application is a web-based e-commerce app where users can browse items, add them to the cart, and purchase them. 
+In this example, we will deploy the demo in the local k8s cluster **kind**. 
+
+## Prerequisites
+* [istio](https://istio.io/)
```
Comment (wonderflow) [inline 7.GoogleCloudPlatform_MicroServices_Demo/README.md:9]: Explain why istio was list here? Do you mean install istio? Then what version is required?

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+### Apply Application 
+
+```shell
+kubectl apply -f online-boutique.yaml
+```
+
+Please wait for a while until all deployments are ready.
+
+```shell
+$ kubectl get pods
```
Comment (wonderflow) [inline 7.GoogleCloudPlatform_MicroServices_Demo/README.md:43]: use `kubectl get deployment` is better

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+
+1. Create an Istio DestinationRule for productcatalogservice.
+```
+kubectl apply -f istio-canary/destinationrule.yaml
+```
+
+2. Deploy ProductCatalog v2. We need open [online-boutique.yaml](./online-boutique.yaml)  and comment the Component productcatalogservice.
+
+``` YAML
+    # - name: productcatalogservice
```
Comment (wonderflow) [inline 7.GoogleCloudPlatform_MicroServices_Demo/README.md:110]: why the yaml is in `#` ?

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+In this demo, we abstracted out 2 types of workload: [microservice](./Definitions/workloads/micrioservice.yaml) and [enhanced-worker](./Definitions/workloads/enhanced-worker.yaml).
+
+1. microservice: microservice describe a workload component Deployment with Service.
+2. enhanced-worker: enhanced-worker describe a long-running, scalable, containerized services that running at backend. They do NOT have network endpoint to receive external network traffic.
+
+Application **boutique** is composed of 11 microservices and 1 worker. microservices provide shopping services, The worker continuously sends requests imitating realistic user shopping flows to the frontend.
+
+![Architecture of microservices](./architecture-diagram.png)
+
+```
```
Comment (wonderflow) [inline 7.GoogleCloudPlatform_MicroServices_Demo/README.md:24]: explain why install these workloads

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+kubectl apply -f Definitions/workloads
+```
+
+### Apply TraitDefinition
+
+```shell
```
Comment (wonderflow) [inline 7.GoogleCloudPlatform_MicroServices_Demo/README.md:30]: same here

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md (hunk 2 above)
Comment (wonderflow) [reply 7.GoogleCloudPlatform_MicroServices_Demo/README.md:43]: other resource like `service`

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+```
+
+3. Uncomment the Component productcatalogservice-v2.
+
+``` YAML
+    - name: productcatalogservice
+      type: microservice
+      settings:
+        version: "v2"
+        image: gcr.io/google-samples/microservices-demo/productcatalogservice:v0.1.3
```
Comment (wonderflow) [inline 7.GoogleCloudPlatform_MicroServices_Demo/README.md:139]: please use a mirror image

**Review by wonderflow — APPROVED** [review]
LGTM

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+2. enhanced-worker: enhanced-worker describe a long-running, scalable, containerized services that running at backend. They do NOT have network endpoint to receive external network traffic.
+
+We register these workloads to Kubevela.
+
+```shell
+kubectl apply -f Definitions/workloads
+```
+
+### Apply TraitDefinition
+We have defined 7 traits to enhance the ability of wordload.
```
Comment (wonderflow) [inline 7.GoogleCloudPlatform_MicroServices_Demo/README.md:33]:
```suggestion
We have defined 7 traits to enhance the ability of workload.
```

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md (hunk 6 above)
Comment (yangsoon) [reply 7.GoogleCloudPlatform_MicroServices_Demo/README.md]: The image has been synchronized to Alibaba Cloud ACR

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md (hunk 4 above)
Comment (yangsoon) [reply 7.GoogleCloudPlatform_MicroServices_Demo/README.md]: Got it.

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md (hunk 5 above)
Comment (yangsoon) [reply 7.GoogleCloudPlatform_MicroServices_Demo/README.md:46]: Got it.

File: 7.GoogleCloudPlatform_MicroServices_Demo/README.md [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+NAME                   TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                                      AGE
+istio-ingressgateway   LoadBalancer   <your ip>       <pending>     15021:30077/TCP,80:31791/TCP,443:32113/TCP,31400:31457/TCP,15443:32035/TCP   12m
+```
+
+We do port forwarding to the ingressgateway of istio.
+
+```
+kubectl port-forward svc/istio-ingressgateway -n istio-system 8080:80
+```
+Then you can access the **Online Boutique** from [http://127.0.0.1:8080/](http://127.0.0.1:8080/).
```
[the remark this answers is not in the document]
Comment (yangsoon) [reply 7.GoogleCloudPlatform_MicroServices_Demo/README.md:159]: OK

**Pull request merged**